svm/src/transaction_processor.rs

```diff
@@ -484,32 +469,65 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         let fee_payer_index = 0;
         validate_fee_payer(
             fee_payer_address,
-            &mut fee_payer_account,
+            &mut loaded_fee_payer.account,
             fee_payer_index,
             error_counters,
             rent_collector,
             fee_details.total_fee(),
         )?;
 
+        // If the nonce has been used in this batch already, we must drop the transaction
+        // This is the same as if it was used in different batches in the same slot
+        // If the nonce account was closed in the batch, we error as if the blockhash didn't validate
+        // We must vaidate the account in case it was reopened, either as a normal system account, or a fake nonce account
```

Comment:
typo:
```suggestion
        // We must validate the account in case it was reopened, either as a normal system account, or a fake nonce account
```